backend-dotnet/MapService/Controllers/MapConfigController.cs

```diff
@@ -0,0 +1,697 @@
+                                               + " This means that you accept the value of X-Control-Header from any request, which is potentially a huge security risk!.");
+                    }
+                    if (!adHandler.RequestComesFromAcceptedIp(HttpContext))
+                    {
+                        return StatusCode(StatusCodes.Status500InternalServerError, "AD authentication does not allow requests from " + remoteIpAddress + ". Aborting.");
+                    }
+
+                    userPrincipalName = adHandler.PickUserNameToUse(Request, userPrincipalName);
+
+                    if (!adHandler.UserIsValid(userPrincipalName) || !AdHandler.UserHasAdAccess(userPrincipalName))
+                    {
+                        return StatusCode(StatusCodes.Status403Forbidden, "Forbidden");
+                    }
+                }
+
+                var appDataFolderPath = MapService.Utility.PathUtility.GetPath("DataContent:Path");
+                _logger.LogInformation("GetLayers DataFolderPath: {appDataFolderPath} ", appDataFolderPath);
+
+                layers = MapConfigHandler.GetLayersAsJsonDocument();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Internal Server Error");
+                return StatusCode(StatusCodes.Status500InternalServerError, "Internal Server Error");
+            }
+
+            return StatusCode(StatusCodes.Status200OK, layers);
+        }
+
+        /// <remarks>
+        /// Fetch contents of a map configuration.
+        /// </remarks>
+        /// <param name="map">The name of the map</param>
+        /// <param name="userPrincipalName">User name that will be supplied to AD. This header can be configured by the administrator to be named something other than X-Control-Header.</param>
+        /// <returns>Returns a map as a JsonObject</returns>
+        /// <response code="200">The map object fetched successfully</response>
+        /// <response code="403">Forbidden</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpGet]
+        [Route("{map}")]
+        [MapToApiVersion("1.0")]
+        [MapToApiVersion("2.0")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerOperation(Tags = new[] { "Admin - Maps and layers" })]
+        public ActionResult GetMap(string map, [FromHeader(Name = "X-Control-Header")] string? userPrincipalName)
+        {
+            JsonDocument mapDocument;
+
+            try
+            {
+                if (AdHandler.AdIsActive)
+                {
+                    var adHandler = new AdHandler(_memoryCache, _logger);
+
+                    string? remoteIpAddress = adHandler.GetRemoteIpAddress(HttpContext);
+                    if (!adHandler.IpRangeRestrictionIsSet())
```

Comment:
If I have IP-range set to null in config and X-Control-Header to null I will never be able to get the map. Today, I don't use any proxy or X-Control-Header. Maybe I got something backwards...........

Reply:
Just had a quick look at the code and i think you are right. If you use windows-authentication we still check the ip-restriction, which should only be done when grabbing the username from the header, right?

Reply:
Yes, just to clarify:
```
"LookupActive": true,
"IdentifyUserWithWindowsAuthentication": true,
"TrustedProxyIPs": null,
"TrustedHeader": null,
```

Reply:
@sweco-semtto Do you have any input on this? Did you test the AD-connection with windows authentication? If so, could you guide @jesade-vbg on how it should be set up?

Reply:
We tested the AD connection using Windows authentication. Correct me if I'm wrong, but I don't see a valid case where TrustedHeader is set to null. I will be more than happy to guide @jesade-vbg on how set this up.

Reply:
We're not fetching the user name from a header-key if we're using Windows authentication?

```c#
public string PickUserNameToUse(HttpRequest request, string? userName)
{
    if (IdentifyUserWithWindowsAuthentication)
        return GetWindowsAuthenticationUserName(request.HttpContext);
    else
        return GetValueFromTrustedHeader(request, userName);
}

public string GetWindowsAuthenticationUserName(HttpContext? httpContext)
{
    if (httpContext?.User?.Identity?.IsAuthenticated == true)
    {
        var userName = httpContext.User.Identity.Name;
        var userNameWithoutDomain = userName.Split('\\').Last();

        _logger.LogInformation("Authenticated user: {0}", userNameWithoutDomain);

        return userNameWithoutDomain;
    }
    else
    {
        _logger.LogInformation("User is not authenticated.");
        return string.Empty;
    }
}
```

I don't get why the trusted header should be set when we're dealing with the Windows authentication approach. Am i missing something? I think the code above suggests that the trusted header can be null in the Windows authentication case.

Reply:
Also, i don't think the nullish `TrustedHeader` is the main problem  but rather why the ip-restriction must be set even when we're using Windows authentication.

Reply:
@sweco-semtto I can't actually see why I would need a header with user and IP-restriction when IIS is exposed at 443 and handling users with windowsAuth. Either way.... I don't have anything to give me any X-Control-Header.